artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -343,6 +347,18 @@ public void onNotification(final Notification notification) {
                   if (binding != null) {
                      // We have a local queue
                      Queue queue = (Queue) binding.getBindable();
+                     //If the client ID represents a client already connected to the server then the server MUST disconnect the existing client.
+                     //Avoid consumers with the same client ID in the cluster appearing at different nodes at the same time
+                     SimpleString protocolName = props.getSimpleStringProperty(ManagementHelper.HDR_PROTOCOL_Name);
```

Comment:
I'm not that familiar with the MQTT protocol, but I can give you an idea: instead of using a SimpleString property, that is costly to be compared/transferred/encoded/decoded why not define a single byte (or short to be more future-proof) property? 
It would be less human-readable, but more cheaper for a machine: in addition, a Short/Byte tends to be pooled on the ranges of -127<->128 AFAIK or something similar, so only the property name + HashMap entry will produce some garbage.

Reply:
Thanks, very nice advice. i will done